app/src/main/java/org/mozilla/focus/activity/MainActivity.kt

```diff
@@ -112,6 +125,33 @@ open class MainActivity : LocaleAwareAppCompatActivity() {
         AppReviewUtils.showAppReview(this)
     }
 
+    private fun recordMultiTabExposure() {
+        val multiTabsExperiment = FocusNimbus.features.tabs
+        if (multiTabsExperiment.value().isMultiTab) {
+            multiTabsExperiment.recordExposure()
+        }
+    }
+
+    private fun setSplashScreenPreDrawListener() {
+        val content: View = findViewById(android.R.id.content)
+        content.viewTreeObserver.addOnPreDrawListener(object : ViewTreeObserver.OnPreDrawListener {
+            override fun onPreDraw(): Boolean {
+                experimentsRequestExpired = settings.experimentsStartFetchingTimestamp?.let { startTimestamp ->
+                    System.currentTimeMillis() - startTimestamp >= REQUEST_TIME_OUT
+                } ?: false
+
+                return if (experimentsRequestExpired) {
+                    components.experiments.applyPendingExperiments()
+                    content.viewTreeObserver.removeOnPreDrawListener(this)
+                    true
+                } else {
+                    false
+                }
+            }
+        }
+        )
+    }
```

Comment:
When is `onPreDraw()` called? I'm not sure I understand what is happening here.

Reply:
https://developer.android.com/guide/topics/ui/splash-screen#:~:text=Starting%20in%20Android%2012%2C%20the,transition%20to%20your%20app%20itself.
_Keep the splash screen on-screen for longer periods
The splash screen is dismissed as soon as your app draws its first frame. If you need to load a small amount of data such as in-app settings from a local disk asynchronously, you can use [ViewTreeObserver.OnPreDrawListener](https://developer.android.com/reference/android/view/ViewTreeObserver.OnPreDrawListener) to suspend the app to draw its first frame._

Reply:
I applied your suggestion https://github.com/mozilla-mobile/focus-android/pull/6618#discussion_r852872902
After 2 seconds applyPendingExperiments() is called(no matter what) and the splash screen is closed